metaseq/tasks/streaming_language_modeling.py

```diff
@@ -266,16 +279,7 @@ def load_dataset(self, split: str, epoch=1, combine=False, **kwargs):
         # shuffles them, then chunks them into blocks of tokens (e.g., 2048).
 
         # determine number of shards for this split
-        shards = {}
-        for shard_id in os.listdir(os.path.join(self.args.data, split)):
-            assert (
-                int(shard_id) not in shards
-            ), f"shard id: {shard_id} not in shards: {shards}"
-            shards[int(shard_id)] = shard_id
-        assert min(shards.keys()) == 0
-        assert max(shards.keys()) == len(shards) - 1
-
-        cur_shard_str = shards[(epoch - 1) % len(shards)]
+        cur_shard_str = self.get_shard_str(epoch, split)
```

Comment:
To confirm: this logic remains unchanged, correct?  The main change here is to move it into a separate function?

Reply:
Correct. Just to reuse this code in the derived class.